cli/cmd/encore/eject.go

```diff
@@ -34,8 +34,20 @@ func init() {
 		},
 	}
 
+	lang, err := appfile.AppLang(p.AppRoot)
+	if err != nil {
+		lang = appfile.LangGo
+	}
+	p.BaseImg = "scratch"
+	if lang == appfile.LangTS {
+		p.BaseImg = "node"
+	}
+
 	dockerEjectCmd.Flags().BoolVarP(&p.Push, "push", "p", false, "push image to remote repository")
-	dockerEjectCmd.Flags().StringVar(&p.BaseImg, "base", "scratch", "base image to build from")
+	dockerEjectCmd.Flags().StringVar(&p.BaseImg, "base", p.BaseImg, "base image to build from")
+	dockerEjectCmd.Flags().StringVar(&p.Goos, "os", p.Goos, "target operating system. One of (linux, darwin, windows)")
+	dockerEjectCmd.Flags().StringVar(&p.Goarch, "arch", p.Goarch, "target architecture. One of (amd64, arm64)")
+	dockerEjectCmd.Flags().BoolVar(&p.CgoEnabled, "cgo", p.CgoEnabled, "enable cgo")
```

Comment:
Should we read this from `encore.app` instead? I guess it's a bit annoying to do so :)

Reply:
Ah, for cgo that makes sense of course. I already read encore.app to find the language (to select default base image). Would you want to be able to specify arch and os in the encore.app too?

Reply:
Nah, arch/os seems more like the concern of a specific invocation of building an image, whereas using cgo is more a property of the app's requirements